server/controllers/offer-type/addOfferType.js

```diff
@@ -1,3 +1,19 @@
-module.exports = (req, res) => {
-  res.send(' add new offer type');
+const { addOfferType } = require('../../database/queries/offer-type/postOfferType.js');
+const yup = require('yup');
+
+module.exports = (req, res, next) => {
+  const name = req.body;
+  const schema = yup.object({
+    name: yup.string().required(),
+  });
+  schema.validate({ name })
+    .then(() => {
+      addOfferType(name)
```

Comment:
you should check if the ```name``` of offer_type is new or not, so you need the query function :
```checkofferTypeName = (name) => {
  const sql = {
    text: 'select * from skill where offer_type.name = $1',
    values: [name],
  };
  return connection.query(sql);
};```

Reply:
@Angham116 good suggestion 👍 
@alaabadra Once you made that change, you could add a test too (which ensures that if you try to post an offer_type with a name that's already taken, the request 400's)

Reply:
yes , i changed it 
thank you tom